src/common/updating/updating.cpp

```diff
@@ -117,39 +117,44 @@ namespace updating
+            constexpr const std::wstring_view required_filename_pattern = updating::INSTALLER_FILENAME_PATTERN;
+            // Desc-sorted by its priority
+            const std::array<std::wstring_view, 2> asset_extensions = { L".exe", L".msi" };
+            for (const auto asset_extension : asset_extensions)
+            {
+                for (auto asset_elem : json_body.GetNamedArray(L"assets"))
                 {
-                    for (auto asset_elem : json_body.GetNamedArray(L"assets"))
+                    auto asset{ asset_elem.GetObjectW() };
+                    std::wstring filename_lower = asset.GetNamedString(L"name", {}).c_str();
+                    std::transform(begin(filename_lower), end(filename_lower), begin(filename_lower), ::towlower);
+
+                    const bool extension_matched = filename_lower.ends_with(asset_extension);
+                    const bool architecture_matched = filename_lower.find(required_architecture) != std::wstring::npos;
+                    const bool filename_matched = filename_lower.find(required_filename_pattern) != std::wstring::npos;
+                    if (extension_matched && architecture_matched && filename_matched)
                     {
-                        auto asset{ asset_elem.GetObjectW() };
-                        std::wstring filename_lower = asset.GetNamedString(L"name", {}).c_str();
-                        std::transform(begin(filename_lower), end(filename_lower), begin(filename_lower), ::towlower);
-
-                        const bool extension_matched = filename_lower.ends_with(asset_extension);
-                        const bool architecture_matched = filename_lower.find(required_architecture) != std::wstring::npos;
-                        const bool filename_matched = filename_lower.find(required_filename_pattern) != std::wstring::npos;
-                        if (extension_matched && architecture_matched && filename_matched)
-                        {
-                            winrt::Windows::Foundation::Uri msi_download_url{ asset.GetNamedString(L"browser_download_url") };
-                            co_return new_version_download_info{ std::move(release_page_uri), new_version.c_str(), std::move(msi_download_url), std::move(filename_lower) };
-                        }
+                        winrt::Windows::Foundation::Uri msi_download_url{ asset.GetNamedString(L"browser_download_url") };
+                        co_return new_version_download_info{ std::move(release_page_uri), new_version.c_str(), std::move(msi_download_url), std::move(filename_lower) };
                     }
                 }
             }
-            else
-            {
-                co_return std::nullopt;
-            }
+        }
+        catch (winrt::hresult_error const& ex)
+        {
+            std::wstring error_message{ strings.GITHUB_NEW_VERSION_CHECK_ERROR };
+            error_message += L' ';
+            error_message += ex.message().c_str();
```

Comment:
Let's not add this error message in the notification, do we know if this gets localized?
Also it might be too generic or too technical for the final user.
Replace the error with the previously suggested message.

Reply:
The message when the internet is disabled is this one: https://user-images.githubusercontent.com/1828123/99543303-16eea700-29c4-11eb-966e-54eaf14d68ce.png, i.e. "The server name or address could not be resolved".

If we use the suggested message, it would be misleading when a user encounters any other problem.

I've tested the error message string on a Russian Windows, it's localized:
![image](https://user-images.githubusercontent.com/1828123/99554988-dc3f3b80-29d0-11eb-8363-56ff7bbac118.png)

Reply:
`The server name or address could not be resolved` isn't really useful for the users.
Let's agree on a fixed message.
I suggest `Failed to connect to the server. Check your network connection or retry later.` because it covers most of the cases and gives a clear suggestion.